tests/components/zwave_js/test_device_action.py

```diff
@@ -0,0 +1,412 @@
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_event_set_value",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "type": "set_value",
+                        "device_id": "fake",
+                        "command_class": 112,
+                        "property": "test",
+                        "value": 1,
+                    },
+                },
+                {
+                    "trigger": {
+                        "platform": "event",
+                        "event_type": "test_event_set_config_parameter",
+                    },
+                    "action": {
+                        "domain": DOMAIN,
+                        "type": "set_config_parameter",
+                        "device_id": "fake",
+                        "parameter": 3,
+                        "bitmask": None,
+                        "subtype": "2-112-0-3 (Beeper)",
+                        "value": 255,
+                    },
+                },
+            ]
+        },
+    )
+
+    clear_lock_usercode = async_mock_service(hass, "zwave_js", "clear_lock_usercode")
+    hass.bus.async_fire("test_event_clear_lock_usercode")
+    await hass.async_block_till_done()
+    assert len(clear_lock_usercode) == 1
+
+    set_lock_usercode = async_mock_service(hass, "zwave_js", "set_lock_usercode")
+    hass.bus.async_fire("test_event_set_lock_usercode")
+    await hass.async_block_till_done()
+    assert len(set_lock_usercode) == 1
+
+    refresh_value = async_mock_service(hass, "zwave_js", "refresh_value")
+    hass.bus.async_fire("test_event_refresh_value")
+    await hass.async_block_till_done()
+    assert len(refresh_value) == 1
+
+    ping = async_mock_service(hass, "zwave_js", "ping")
+    hass.bus.async_fire("test_event_ping")
+    await hass.async_block_till_done()
+    assert len(ping) == 1
+
+    set_value = async_mock_service(hass, "zwave_js", "set_value")
+    hass.bus.async_fire("test_event_set_value")
+    await hass.async_block_till_done()
+    assert len(set_value) == 1
+
+    set_config_parameter = async_mock_service(hass, "zwave_js", "set_config_parameter")
+    hass.bus.async_fire("test_event_set_config_parameter")
+    await hass.async_block_till_done()
+    assert len(set_config_parameter) == 1
+
+
+async def test_get_action_capabilities(
+    hass: HomeAssistant,
+    client: Client,
+    climate_radio_thermostat_ct100_plus: Node,
+    integration: ConfigEntry,
+):
+    """Test we get the expected action capabilities."""
+    node = climate_radio_thermostat_ct100_plus
+    dev_reg = device_registry.async_get(hass)
+    device = device_registry.async_entries_for_config_entry(
+        dev_reg, integration.entry_id
+    )[0]
+
+    # Test refresh_value
+    capabilities = await device_action.async_get_action_capabilities(
```

Comment:
Can we avoid using details of the platform in the tests?

Reply:
Can you clarify what you mean here? I used the same pattern I used in the other device automation tests so depending on your response I may need to update those as well

Reply:
If possible it's better to start the test with a core interface, eg by setting up the automation integration, or a common helper function. I don't know if we have a good alternative.

Reply:
I think in this case there isn't because this is something the frontend would typically call